### ISSUE TITLE
KeepassXC-Browser no longer detecting credential fields or auto filling

### ISSUE THREAD

**Issue by jamz216** [keepassxreboot/keepassxc-browser#300]
## Expected Behavior
A site saved in my KeepassXC loads and the username/password are auto-filled by KeepassXC-Browser. 

## Current Behavior
I was having trouble with some but not all sites last week but I have since downloaded a new copy of my database, there has been an updated to chrome, and I had to clear cache/cookies for an unrelated reason and now KeepassXC-Browser will not detect/fill in the login in for for any site. I've tried opening the site from within the database (with Cmd + U) and loading the site separately. I have tried hitting "re-detect credential fields" and have tried "choose custom fields for this page". 

## Debug info
KeePassXC - Version 2.3.4
Revision: 6fe821c

Libraries:
- Qt 5.9.3
- libgcrypt 1.8.2

Operating system: macOS High Sierra (10.13)
CPU architecture: x86_64
Kernel: darwin 17.7.0

Enabled extensions:
- Auto-Type
- Browser Integration
- Legacy Browser Integration (KeePassHTTP)
- SSH Agent
- YubiKey

KeePassXC-Browser Version: 1.2.0
Operating system: Mac 
Browser: Chrome

Thank you

**Comment by varjolintu** [comment]
What's the Chrome version you are using? Can you open the Inspector via right mouse click (and select Inspect) and see if the console has any error messages when you browse to a page and try the fill?

**Comment by whisdol** [comment]
I had the same issue in Chrome and Firefox (Nightly) on macOS and Windows 10 recently and was unable to completely pin-point it to single cause. Things I tried that eventually got it working again:
- Remove all sites from the site preferences tab.
- Ensure "Automatically retrieve credentials." is checked.

**Comment by varjolintu** [comment]
That actually might be the cause. If there's a site in Site Preferences without the last `/` in the URL, content scripts might not work at all. It's already fixed in the next release. But please make sure every URL that is just a plain hostname, e.g. `https://google.com/` has the last slash.

**Comment by varjolintu** [comment]
In KeePassXC 2.3.4 the title matching was removed (https://github.com/keepassxreboot/keepassxc/pull/2055) which broke the subdomain matching (if it ever worked because it relied to titles, PR: https://github.com/keepassxreboot/keepassxc/pull/2253). And also the scheme matching was not working correctly (https://github.com/keepassxreboot/keepassxc/pull/2232). The next KeePassXC release will fix the most of these problems.

The only workaround is to use exact URL's with identical schemes.

In short, the browser extension is not more picky, KeePassXC is. The matching was way too loose in the previous versions.

**Comment by apiraino** [comment]
thanks @varjolintu for the explaination, much appreciated.

Looking forward for the new release, then :+1:

**Comment by jamz216** [comment]
> What's the Chrome version you are using? Can you open the Inspector via right mouse click (and select Inspect) and see if the console has any error messages when you browse to a page and try the fill?

I went to zappos.com and this message was displayed as an error message (for a different page that I have been to)
Uncaught (in promise) TypeError: https://www.fpntc.org is not a valid MatchPattern
    at matchPatternToRegExp (global.js:48)
    at siteMatch (global.js:91)
    at browser.runtime.sendMessage.then (keepassxc-browser.js:1431)


I will also give the other suggested solutions a try and report back, thank you everyone!

**Comment by varjolintu** [comment]
@jamz216 Then [https://github.com/keepassxreboot/keepassxc-browser/issues/300#issuecomment-421274311](this) comment is the solution for your problem.

**Comment by jamz216** [comment]
I will report back in a few days when I have a chance to see which sites aren't working, thank you!

**Comment by jamz216** [comment]
Was still not working but I've just been going in to the keepass database now and adding https:// to every website that doesn't work and that seems to have fixed the issue. Hopefully this will be fixed in the next release to work as it did before. Thanks everyone for the help!

**Comment by Gunni** [comment]
I had this problem too, and there were two domains in extension settings under site preferences. Neither ended in a `/` and when removed the extension started working again.
